Features/Intel/UserInterface/VirtualKeyboardFeaturePkg/VirtualKeyboardDxe/Keyboard.c

```diff
@@ -883,33 +883,48 @@ VkApiStart (
 
   KeyData.KeyState.KeyToggleState = 0;
   KeyData.KeyState.KeyShiftState  = 0;
-  KeyData.Key.ScanCode            = SCAN_ESC;
   KeyData.Key.UnicodeChar         = CHAR_NULL;
   NotifyHandle                    = NULL;
-  Status = SimpleEx->RegisterKeyNotify (
-                       SimpleEx,
-                       &KeyData,
-                       VkNotifyKeyCallback,
-                       &NotifyHandle
-                       );
-  if (EFI_ERROR (Status)) {
-    DEBUG ((DEBUG_VK_ROUTINE_ENTRY_EXIT | DEBUG_ERROR, "ERROR - Failed to register 'Esc', Status: %r\n", Status));
-    goto Error;
+
+  for (KeyData.Key.ScanCode = SCAN_UP; KeyData.Key.ScanCode <= SCAN_ESC; KeyData.Key.ScanCode++) {
+    Status = SimpleEx->RegisterKeyNotify (
+                         SimpleEx,
+                         &KeyData,
+                         VkNotifyKeyCallback,
+                         &NotifyHandle
+                         );
+    if (EFI_ERROR (Status)) {
+      DEBUG ((
+        DEBUG_VK_ROUTINE_ENTRY_EXIT | DEBUG_ERROR,
+        "ERROR - Failed to register '%c', Status: %r\n",
+        KeyData.Key.ScanCode,
+        Status
+        ));
+      break;
+    }
   }
 
   KeyData.KeyState.KeyToggleState = 0;
   KeyData.KeyState.KeyShiftState  = 0;
-  KeyData.Key.ScanCode            = SCAN_NULL;
-  KeyData.Key.UnicodeChar         = CHAR_CARRIAGE_RETURN;
-  Status = SimpleEx->RegisterKeyNotify (
-                       SimpleEx,
-                       &KeyData,
-                       VkNotifyKeyCallback,
-                       &NotifyHandle
-                       );
-  if (EFI_ERROR (Status)) {
-    DEBUG ((DEBUG_VK_ROUTINE_ENTRY_EXIT | DEBUG_ERROR, "ERROR - Failed to register 'Enter', Status: %r\n", Status));
-    goto Error;
+  KeyData.Key.UnicodeChar         = CHAR_NULL;
+  NotifyHandle                    = NULL;
+
+  for (KeyData.Key.UnicodeChar = CHAR_BACKSPACE; KeyData.Key.UnicodeChar <= CHAR_CARRIAGE_RETURN; KeyData.Key.UnicodeChar++) {

```

Comment:
Can we keep the CHAR_CARRIAGE_RETURN separately as it was before, and keep the current loop till CHAR_LINEFEED,
because Unicode characters 0x000B and 0x00C are not defined.

Reply:
thanks for your review. Are you sure WRT comment? Another reviewer asked me to absorb CHAR_CARRIEGE_RETURN into the loop. 
![image](https://github.com/user-attachments/assets/093e9e15-f649-4fce-9428-b7f7966413c1)